devito/types/misc.py

```diff
@@ -342,3 +342,17 @@ def closing(self):
 A wildcard for use in the RHS of Eqs that encode some kind of semantics
 (e.g., a synchronization operation) but no computation.
 """
+
+
+# *** CXX support types
+
+auto = CustomDtype('auto')
```

Comment:
I get why it is done like that, but it's not really a type. It's a return qualifier. I'm not sure how else it should be done; maybe add a TODO or issue for it.

Reply:
done, pushing updated version now